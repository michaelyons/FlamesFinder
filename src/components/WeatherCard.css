.current-weather-card {
  display: flex;
  flex-direction: column;

  padding-bottom: 0.5rem;
  padding-top: 0.5rem;

  color: #020202;
  background-color: rgba(250, 250, 250, 0.7);

  -webkit-box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);
  box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);
}

p .current-weather-card {
  margin: 0;
}

.ten-hour-container {
  border: 2px solid black;
  width: 100vw;
  display: flex;
  flex-direction: row;
  align-content: center;
  overflow-x: scroll;
  margin: auto;
}

.hour-weather-card {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 10rem;
  padding: 0.5rem 1.5rem 0.5rem 1.5rem;
  color: #020202;
  background-color: rgba(250, 250, 250, 0.7);

  -webkit-box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);
  box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);
}

.ten-day-container {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
}

.day-weather-card {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;

  color: #020202;
  background-color: rgba(250, 250, 250, 0.7);

  -webkit-box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);
  box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);
}
