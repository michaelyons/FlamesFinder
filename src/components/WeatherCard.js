import React, { Component } from 'react';
const uuidv1 = require('uuid/v1');
export class WeatherCard extends Component {
  render() {
    let displayCurrentWeather;

    let displayTenHourWeather;

    let displayTenDayWeather;

    if (!this.props.currentWeather) {
      displayCurrentWeather = <p className="loading-fetch">Loading...</p>;
    } else {
      displayCurrentWeather = (
        <section>
          <p>{this.props.currentWeather.location}</p>
          <p>{this.props.currentWeather.currentTemp}</p>
          <p>{this.props.currentWeather.humidity}</p>
          <p>{this.props.currentWeather.highTemp}</p>
          <p>{this.props.currentWeather.lowTemp}</p>
          <p>{this.props.currentWeather.windSpeed}</p>
          <p>{this.props.currentWeather.sunrise}</p>
          <p>{this.props.currentWeather.sunset}</p>
          <p>{this.props.currentWeather.windDirection}</p>
          <p>{this.props.currentWeather.visibility}</p>
          <p>{this.props.currentWeather.currentConditions}</p>
          <p>{this.props.currentWeather.latitude}</p>
          <p>{this.props.currentWeather.longitude}</p>
        </section>
      );
    }

    if (!this.props.tenHourWeather) {
      displayTenHourWeather = <p className="loading-fetch">Loading...</p>;
    } else {
      displayTenHourWeather = this.props.tenHourWeather.map(weather1 => {
        return (
          <section key={uuidv1()}>
            <p>{`${weather1.time}`}</p>
            <p>{`${weather1.temp}`}</p>
            <p>{`${weather1.feelsLike}`}</p>
            <p>{`${weather1.averageHumidity}`}</p>
            <p>{`${weather1.condition}`}</p>
            <p>{`${weather1.averageWind}`}</p>
            <p>{`${weather1.uvIndex}`}</p>
          </section>
        );
      });
    }
    if (!this.props.tenDayWeather) {
      displayTenDayWeather = <p className="loading-fetch">Loading...</p>;
    } else {
      displayTenDayWeather = this.props.tenDayWeather.map(weather2 => {
        return (
          <section key={uuidv1()}>
            <p>{`${weather2.day}`}</p>
            <p>{`${weather2.date}`}</p>
            <p>{`${weather2.high}`}</p>
            <p>{`${weather2.low}`}</p>
            <p>{`${weather2.conditions}`}</p>
            <p>{`${weather2.averageWind}`}</p>
          </section>
        );
      });
    }
    return (
      <div>
        <button>Click for Nearby Campgrounds</button>
        <div>{displayCurrentWeather}</div>
        <div>{displayTenHourWeather}</div>
        <div>{displayTenDayWeather}</div>
      </div>
    );
  }
}
