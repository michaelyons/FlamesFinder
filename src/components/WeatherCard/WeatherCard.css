.current-weather-card {
  display: flex;
  flex-direction: column;
  color: aliceblue;
  font-size: 1.7rem;
  margin-bottom: 1rem;
  letter-spacing: 0.1rem;
  font-weight: bold;
}

.current-weather-card p {
  margin: 0;
}

.ten-hour-container {
  width: 100vw;
  display: flex;
  flex-direction: row;
  align-content: center;
  overflow-x: scroll;
  margin: auto;
  font-size: 1.6rem;
  border-top: 3px solid aliceblue;
}

.hour-weather-card {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 10rem;
  padding: 0.5rem 2rem 0.5rem 2rem;
  color: aliceblue;
  border-right: 3px solid aliceblue;
  letter-spacing: 0.1rem;
  font-weight: bold;
}

.hour-weather-card p {
  margin: 0.5rem;
}

.ten-day-container {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  margin-bottom: 3rem;
  border-top: 3px solid aliceblue;
}

.day-weather-card {
  font-size: 1.6rem;
  font-weight: bold;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  padding-bottom: 0.25rem;
  padding-top: 0.5rem;
  color: aliceblue;
  border-bottom: 3px solid aliceblue;
  letter-spacing: 0.1rem;
}
