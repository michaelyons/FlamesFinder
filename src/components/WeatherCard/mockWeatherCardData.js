export const currentWeatherMock = {
  location: 'Dove Valley',
  currentWeather: 'Current Weather',
  currentTemp: '80°F',
  humidity: 'Humidity 25%',
  highTemp: 'High 85°F',
  lowTemp: 'Low 78°F',
  windSpeed: 'Wind Speed 13.87 MPH',
  sunrise: 'Sunrise 6:44:56 AM',
  sunset: 'Sunset 6:59:55 PM',
  windDirection: 'Wind Direction NE',
  visibility: 'Visibility 10.00 Miles',
  currentConditions: 'Broken Clouds'
};

export const tenHourWeatherMock = [
  {
    time: '1:00 PM',
    temp: '87°F',
    feelsLike: '87°F',
    averageHumidity: '15%',
    condition: 'Mostly Cloudy',
    averageWind: 'ENE 12',
    uvIndex: '5/10',
    icon: 'mostlycloudy'
  },
  {
    time: '2:00 PM',
    temp: '88°F',
    feelsLike: '88°F',
    averageHumidity: '15%',
    condition: 'Overcast',
    averageWind: 'ENE 13',
    uvIndex: '4/10',
    icon: 'cloudy'
  },
  {
    time: '3:00 PM',
    temp: '87°F',
    feelsLike: '87°F',
    averageHumidity: '17%',
    condition: 'Overcast',
    averageWind: 'E 13',
    uvIndex: '2/10',
    icon: 'cloudy'
  },
  {
    time: '4:00 PM',
    temp: '84°F',
    feelsLike: '84°F',
    averageHumidity: '20%',
    condition: 'Overcast',
    averageWind: 'E 12',
    uvIndex: '2/10',
    icon: 'cloudy'
  },
  {
    time: '5:00 PM',
    temp: '82°F',
    feelsLike: '82°F',
    averageHumidity: '25%',
    condition: 'Chance of a Thunderstorm',
    averageWind: 'ENE 11',
    uvIndex: '1/10',
    icon: 'chancetstorms'
  },
  {
    time: '6:00 PM',
    temp: '79°F',
    feelsLike: '79°F',
    averageHumidity: '30%',
    condition: 'Chance of a Thunderstorm',
    averageWind: 'ENE 13',
    uvIndex: '0/10',
    icon: 'chancetstorms'
  },
  {
    time: '7:00 PM',
    temp: '75°F',
    feelsLike: '75°F',
    averageHumidity: '36%',
    condition: 'Chance of a Thunderstorm',
    averageWind: 'NE 9',
    uvIndex: '0/10',
    icon: 'chancetstorms'
  },
  {
    time: '8:00 PM',
    temp: '72°F',
    feelsLike: '72°F',
    averageHumidity: '41%',
    condition: 'Chance of a Thunderstorm',
    averageWind: 'N 8',
    uvIndex: '0/10',
    icon: 'chancetstorms'
  },
  {
    time: '9:00 PM',
    temp: '68°F',
    feelsLike: '68°F',
    averageHumidity: '53%',
    condition: 'Chance of a Thunderstorm',
    averageWind: 'NNW 9',
    uvIndex: '0/10',
    icon: 'chancetstorms'
  },
  {
    time: '10:00 PM',
    temp: '65°F',
    feelsLike: '65°F',
    averageHumidity: '63%',
    condition: 'Chance of a Thunderstorm',
    averageWind: 'ENE 7',
    uvIndex: '0/10',
    icon: 'chancetstorms'
  }
];

export const tenDayWeatherMock = [
  {
    day: 'Wednesday',
    date: '9/19/2018',
    high: '87°F',
    low: '59°F',
    icon: 'chancetstorms',
    conditions: 'Chance of a Thunderstorm',
    averageWind: 'ENE 13 MPH'
  },
  {
    day: 'Thursday',
    date: '9/20/2018',
    high: '81°F',
    low: '49°F',
    icon: 'partlycloudy',
    conditions: 'Partly Cloudy',
    averageWind: 'NNW 11 MPH'
  },
  {
    day: 'Friday',
    date: '9/21/2018',
    high: '72°F',
    low: '50°F',
    icon: 'clear',
    conditions: 'Clear',
    averageWind: 'SE 18 MPH'
  },
  {
    day: 'Saturday',
    date: '9/22/2018',
    high: '82°F',
    low: '54°F',
    icon: 'clear',
    conditions: 'Clear',
    averageWind: 'SE 10 MPH'
  },
  {
    day: 'Sunday',
    date: '9/23/2018',
    high: '84°F',
    low: '54°F',
    icon: 'clear',
    conditions: 'Clear',
    averageWind: 'SE 9 MPH'
  },
  {
    day: 'Monday',
    date: '9/24/2018',
    high: '79°F',
    low: '48°F',
    icon: 'clear',
    conditions: 'Clear',
    averageWind: 'N 10 MPH'
  },
  {
    day: 'Tuesday',
    date: '9/25/2018',
    high: '67°F',
    low: '44°F',
    icon: 'clear',
    conditions: 'Clear',
    averageWind: 'ESE 11 MPH'
  },
  {
    day: 'Wednesday',
    date: '9/26/2018',
    high: '66°F',
    low: '42°F',
    icon: 'partlycloudy',
    conditions: 'Partly Cloudy',
    averageWind: 'E 10 MPH'
  },
  {
    day: 'Thursday',
    date: '9/27/2018',
    high: '71°F',
    low: '43°F',
    icon: 'clear',
    conditions: 'Clear',
    averageWind: 'SE 10 MPH'
  },
  {
    day: 'Friday',
    date: '9/28/2018',
    high: '70°F',
    low: '44°F',
    icon: 'clear',
    conditions: 'Clear',
    averageWind: 'ESE 10 MPH'
  }
];
