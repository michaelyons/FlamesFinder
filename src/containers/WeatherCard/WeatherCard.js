import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { allWeatherData } from '../../helper/apiCalls';
import { addCurrentWeather } from '../../actions/weatherActions';

export class WeatherCard extends Component {
  constructor() {
    super();
    this.state = {
      errors: ''
    };
  }

  populateWeather = async () => {
    try {
      const currentLocationWeather = await allWeatherData();
      this.props.addCurrentWeather(currentLocationWeather);
    } catch (error) {
      this.setState({
        errors: error.message
      });
    }
  };

  render() {
    return <div />;
  }
}

WeatherCard.propTypes = {
  addCurrentWeather: PropTypes.func.isRequired
};

export const mapDispatchToProps = dispatch => ({
  addCurrentWeather: weather => dispatch(addCurrentWeather(weather))
});

export default connect(
  null,
  mapDispatchToProps
)(WeatherCard);
