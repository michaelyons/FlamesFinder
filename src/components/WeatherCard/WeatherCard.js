import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../WeatherCard/WeatherCard.css';

const uuidv1 = require('uuid/v1');

export class WeatherCard extends Component {
  render() {
    let displayCurrentWeather1;
    let displayCurrentWeather2;
    let displayTenHourWeather;
    let displayTenDayWeather;

    if (!this.props.currentWeather) {
      displayCurrentWeather1 = <p className="loading-fetch">Loading...</p>;
    } else {
      displayCurrentWeather1 = (
        <section className="current-weather-card">
          <p>Current Weather</p>
          <p>{this.props.currentWeather.currentTemp}</p>
          <p>{this.props.currentWeather.currentConditions}</p>
        </section>
      );
      displayCurrentWeather2 = (
        <section>
          <p>Humidity: {this.props.currentWeather.humidity}</p>
          <p>High: {this.props.currentWeather.highTemp}</p>
          <p>Low: {this.props.currentWeather.lowTemp}</p>
          <p>Wind Speed: {this.props.currentWeather.windSpeed}</p>
          <p>Sunrise: {this.props.currentWeather.sunrise}</p>
          <p>Sunset: {this.props.currentWeather.sunset}</p>
          <p>Wind Direction: {this.props.currentWeather.windDirection}</p>
          <p>Visibility: {this.props.currentWeather.visibility}</p>
        </section>
      );
    }

    if (!this.props.tenHourWeather) {
      displayTenHourWeather = <p className="loading-fetch">Loading...</p>;
    } else {
      displayTenHourWeather = this.props.tenHourWeather.map(weather1 => {
        return (
          <section key={uuidv1()} className="hour-weather-card">
            <p>{`${weather1.time}`}</p>
            <p>{`${weather1.temp}`}</p>
            <p>{`${weather1.condition}`}</p>
          </section>
        );
      });
    }
    if (!this.props.tenDayWeather) {
      displayTenDayWeather = <p className="loading-fetch">Loading...</p>;
    } else {
      displayTenDayWeather = this.props.tenDayWeather.map(weather2 => {
        return (
          <section key={uuidv1()} className="day-weather-card">
            <p>{`${weather2.day}`}</p>
            <p>{`${weather2.date}`}</p>
            <p>{`${weather2.high}`}</p>
            <p>{`${weather2.low}`}</p>
            <p>{`${weather2.conditions}`}</p>
          </section>
        );
      });
    }
    return (
      <div>
        <div>{displayCurrentWeather1}</div>
        <div className="ten-hour-container">{displayTenHourWeather}</div>
        <div className="ten-day-container">{displayTenDayWeather}</div>
        <div>{displayCurrentWeather2}</div>
      </div>
    );
  }
}

WeatherCard.propTypes = {
  currentWeather: PropTypes.object,
  tenHourWeather: PropTypes.array,
  tenDayWeather: PropTypes.array
};
