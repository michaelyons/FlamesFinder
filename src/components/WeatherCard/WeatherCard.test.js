import React from 'react';
import { shallow } from 'enzyme';

import WeatherCard from './WeatherCard';

describe('WeatherCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<WeatherCard />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
