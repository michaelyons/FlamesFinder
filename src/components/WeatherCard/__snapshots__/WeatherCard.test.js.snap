// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <p
                    className="loading-fetch"
          >
                    Loading...
          </p>
</div>,
        <div
          className="ten-hour-container"
>
          <p
                    className="loading-fetch"
          >
                    Loading...
          </p>
</div>,
        <div
          className="ten-day-container"
>
          <p
                    className="loading-fetch"
          >
                    Loading...
          </p>
</div>,
        <div />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="loading-fetch"
>
            Loading...
</p>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
            "className": "loading-fetch",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="loading-fetch"
>
            Loading...
</p>,
          "className": "ten-hour-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
            "className": "loading-fetch",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="loading-fetch"
>
            Loading...
</p>,
          "className": "ten-day-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
            "className": "loading-fetch",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <p
                        className="loading-fetch"
            >
                        Loading...
            </p>
</div>,
          <div
            className="ten-hour-container"
>
            <p
                        className="loading-fetch"
            >
                        Loading...
            </p>
</div>,
          <div
            className="ten-day-container"
>
            <p
                        className="loading-fetch"
            >
                        Loading...
            </p>
</div>,
          <div />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="loading-fetch"
>
              Loading...
</p>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Loading...",
              "className": "loading-fetch",
            },
            "ref": null,
            "rendered": "Loading...",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="loading-fetch"
>
              Loading...
</p>,
            "className": "ten-hour-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Loading...",
              "className": "loading-fetch",
            },
            "ref": null,
            "rendered": "Loading...",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="loading-fetch"
>
              Loading...
</p>,
            "className": "ten-day-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Loading...",
              "className": "loading-fetch",
            },
            "ref": null,
            "rendered": "Loading...",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
